spec/intelligent_foods_spec.rb

```diff
@@ -129,15 +129,45 @@
 
   describe "#client" do
     it "returns an instance of the api client" do
-      IntelligentFoods.configure do |config|
-        config.client_id = "abc123"
-        config.client_secret = "xyz890"
-      end
-
-      result = IntelligentFoods.client
+      result = configure_client(id: "abc123", secret: "xyz890")
 
       expect(result.id).to eq("abc123")
       expect(result.secret).to eq("xyz890")
     end
+
+    context "when the client is refreshed" do
+      it "forces a new client to be instantiated" do
+        configure_client(id: "abc123", secret: "xyz890")
+        allow(IntelligentFoods::ApiClient).to receive(:new)
+        IntelligentFoods.refresh_client
+
+        IntelligentFoods.client
+
+        expect(IntelligentFoods::ApiClient).to have_received(:new)
+      end
+
+      it "returns a new client with new attributes" do
+        configure_client(id: "abc123", secret: "xyz890")
+        IntelligentFoods.configure do |config|
+          config.client_id = "id123"
+          config.client_secret = "secret123"
+        end
+        IntelligentFoods.refresh_client
+
+        result = IntelligentFoods.client
+
+        expect(result.id).to eq("id123")
+      end
+    end
   end
+
+  def configure_client(id:, secret:)
+    IntelligentFoods.configure do |config|
+      config.client_id = id
+      config.client_secret = secret
+    end
+    IntelligentFoods.refresh_client
```

Comment:
should refreshing the client be part of the configure method? we should probably always refresh when we configure?

Reply:
you should remove this line now.. it's unnecessary, right?